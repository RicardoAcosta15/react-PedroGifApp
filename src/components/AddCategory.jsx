import { useState } from "react"
import PropTypes from 'prop-types'


export const AddCategory = ({onNewCategory}) =>{

    const [inputValue, setInputValue] = useState('')


    //Permite que el input este emitiendo el valor segun 
    //se valla escribiendo
    const ChangeValue = ({target}) =>{
        
        setInputValue(target.value)
    }

    const onSubmit = (event) =>{
        event.preventDefault();
        console.log(inputValue.trim());
        if(inputValue.trim().length <= 1) return;

        // addCategory(categories => [inputValue, ...categories])
        onNewCategory(inputValue)
        setInputValue('')
    }



    return (

        <form onSubmit={onSubmit}>

            <input
                type="text"
                placeholder="Buscar gifs"
                value={inputValue}
                onChange={ChangeValue}
            />

        </form>
    )
}

AddCategory.propTypes ={
    onNewCategory: PropTypes.func.isRequired
}