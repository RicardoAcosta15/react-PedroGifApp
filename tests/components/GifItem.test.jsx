// eslint-disable-next-line no-unused-vars
import { render,screen,getByText } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

/* eslint-disable no-undef */
describe('Pruebas en <GifItem/>', ()=>{
    const title ='Buenas buenas'
    const url = 'hhtps://one-punch.com'

    test('debe hacer match con el snapshot', ()=>{

       const {container}= render(<GifItem title={title} url ={url} />)
       expect(container).toMatchSnapshot()
    });

    test('debe mostrar la imagen con el URL y el ALT indicado', ()=>{
        render(<GifItem title={title} url ={url} />);
        //screen.debug()
        const {src} = screen.getByRole('img')
        expect(src).toBe(url);
    })

    

    
});