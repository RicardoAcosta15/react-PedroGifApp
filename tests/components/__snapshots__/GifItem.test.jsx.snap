// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem/> debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Buenas buenas"
      src="hhtps://one-punch.com"
    />
    <p>
      Buenas buenas
    </p>
  </div>
</div>
`;
